lib/assessment.js

```diff
@@ -32,14 +44,26 @@ var any = function any(array, callback, thisArg) {
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 var select = function select(array, callback, thisArg) {
-
+  var newArray = [];
+  array.forEach(function(currentValue,index, array){
+    if(callback(currentValue,index, array)){
+      newArray.push(array[index]);
+    }
+  })
+  return newArray;
 };
 
 // Question 5
 // Write a function that returns the first element
 // for which invoking the callback returns a truthy value
 var find = function find(array, callback, thisArg) {
-
+  var result;
+  array.forEach(function(currentValue,index, array){
+    if (callback(currentValue,index, array)){
+      return result = array[index];
+    }
+  }
+  return result;
```

Comment:
Maybe `return -1` for the last line, so that it gives back -1 if no match is found?

That's being nit-picky, though. You got this.